doc/explanation/mental-model.rst

```diff
@@ -0,0 +1,175 @@
+The Dune Mental Model
+=====================
+
+It is not strictly necessary to understand Dune's underlying model to use it;
+but knowing how it works under the hood will help understand some errors and
+what's possible with Dune.
+
+.. note::
+
+   This document is a simplification of the reality: the actual rules might be
+   different, it does not touch rule loading and glosses over how caching
+   works, but should be a useful tool to build an understanding of Dune.
+
+How Dune Works
+--------------
+
+The building brick of Dune is the *rule*:
+
+    A *rule* reads *dependencies* and writes *targets* by executing an *action*.
+
+When ``dune build`` is executed, it will first read ``dune`` files to determine
+the rules that apply to the project. Once it has done this, it will determine
+what actions it needs to execute to build the required targets.
+
+An Example
+----------
+
+Let's take the following example.
+
+- there's a CLI tool written in OCaml.
+- it has some build-time configuration stored in ``config.json``.
+- it has an integration test which consists in running in it with
```

Comment:
```suggestion
- it has an integration test which consists of running in it with
```

Reply:
I believe that "consist of" defined the components of something , while "consist in" introduces a definition?
Maybe
"it has an integration test, in which the tool is executed with `testdata.txt` as input"?